ethereum/rpc/backend/backend.go

```diff
@@ -129,10 +134,15 @@ func (e *EVMBackend) GetBlockByNumber(blockNum types.BlockNumber, fullTx bool) (
 func (e *EVMBackend) GetBlockByHash(hash common.Hash, fullTx bool) (map[string]interface{}, error) {
 	resBlock, err := e.clientCtx.Client.BlockByHash(e.ctx, hash.Bytes())
 	if err != nil {
-		e.logger.Warningf("BlockByHash failed for %s", hash.Hex())
+		e.logger.WithError(err).Warningf("BlockByHash failed for %s", hash.Hex())
```

Comment:
@fedekunze I notice here we are using `Warning` log level, wonder if there's any convention when to use warning and debug?

I am leaving the level as it is for now.

Reply:
yes, I created issue #262 to fix it

Reply:
```suggestion
		e.logger.WithError(err).Warningf("BlockByHash block not found", "hash", hash.Hex())
```